.buildkite/scripts/steps/build_and_test.sh

```diff
@@ -109,3 +112,5 @@ if [[ -z "$CPP_CROSS_COMPILE" ]] ; then
   find . -path  "*/**/ml_test_*.out" -o -path "*/**/*.junit" | xargs tar cvzf ${TEST_RESULTS_ARCHIVE}
   buildkite-agent artifact upload "${TEST_RESULTS_ARCHIVE}"
```

Comment:
If a macOS test fails on line 101 then the script will exit there and not upload the test artifacts here.

Also, on line 106 we probably _don't_ want to upload a Linux build if the tests were run and failed.